In [4]:
import numpy as np
import pandas as pd
import matplotlib as plt
%matplotlib inline
import seaborn as sns
import sklearn as sk

In [6]:
df = pd.read_csv('~/module1project/dsc-v2-mod1-final-project-dc-ds-career-042219/kc_house_data.csv')
df.head()

id        date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  10/13/2014  221900.0         3       1.00         1180   
1  6414100192   12/9/2014  538000.0         3       2.25         2570   
2  5631500400   2/25/2015  180000.0         2       1.00          770   
3  2487200875   12/9/2014  604000.0         4       3.00         1960   
4  1954400510   2/18/2015  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
0      5650     1.0         NaN   0.0     ...          7        1180   
1      7242     2.0         0.0   0.0     ...          7        2170   
2     10000     1.0         0.0   0.0     ...          6         770   
3      5000     1.0         0.0   0.0     ...          7        1050   
4      8080     1.0         0.0   0.0     ...          8        1680   

   sqft_basement yr_built  yr_renovated  zipcode      lat     long  \
0            0.0     1955           0.0    98178  47.5112 -122.257   
1          400.0     1951        1991.0    98125  47.7210 -122.319   
2            0.0     1933           NaN    98028  47.7379 -122.233   
3          910.0     1965           0.0    98136  47.5208 -122.393   
4            0.0     1987           0.0    98074  47.6168 -122.045   

   sqft_living15  sqft_lot15  
0           1340        5650  
1           1690        7639  
2           2720        8062  
3           1360        5000  
4           1800        7503  

[5 rows x 21 columns]

In [7]:
df.shape

(21597, 21)

In [9]:
df.describe().T

count          mean           std           min  \
id             21597.0  4.580474e+09  2.876736e+09  1.000102e+06   
price          21597.0  5.402966e+05  3.673681e+05  7.800000e+04   
bedrooms       21597.0  3.373200e+00  9.262989e-01  1.000000e+00   
bathrooms      21597.0  2.115826e+00  7.689843e-01  5.000000e-01   
sqft_living    21597.0  2.080322e+03  9.181061e+02  3.700000e+02   
sqft_lot       21597.0  1.509941e+04  4.141264e+04  5.200000e+02   
floors         21597.0  1.494096e+00  5.396828e-01  1.000000e+00   
waterfront     19221.0  7.595859e-03  8.682485e-02  0.000000e+00   
view           21534.0  2.338627e-01  7.656862e-01  0.000000e+00   
condition      21597.0  3.409825e+00  6.505456e-01  1.000000e+00   
grade          21597.0  7.657915e+00  1.173200e+00  3.000000e+00   
sqft_above     21597.0  1.788597e+03  8.277598e+02  3.700000e+02   
yr_built       21597.0  1.971000e+03  2.937523e+01  1.900000e+03   
yr_renovated   17755.0  8.363678e+01  3.999464e+02  0.000000e+00   
zipcode        21597.0  9.807795e+04  5.351307e+01  9.800100e+04   
lat            21597.0  4.756009e+01  1.385518e-01  4.715590e+01   
long           21597.0 -1.222140e+02  1.407235e-01 -1.225190e+02   
sqft_living15  21597.0  1.986620e+03  6.852305e+02  3.990000e+02   
sqft_lot15     21597.0  1.275828e+04  2.727444e+04  6.510000e+02   

                        25%           50%           75%           max  
id             2.123049e+09  3.904930e+09  7.308900e+09  9.900000e+09  
price          3.220000e+05  4.500000e+05  6.450000e+05  7.700000e+06  
bedrooms       3.000000e+00  3.000000e+00  4.000000e+00  3.300000e+01  
bathrooms      1.750000e+00  2.250000e+00  2.500000e+00  8.000000e+00  
sqft_living    1.430000e+03  1.910000e+03  2.550000e+03  1.354000e+04  
sqft_lot       5.040000e+03  7.618000e+03  1.068500e+04  1.651359e+06  
floors         1.000000e+00  1.500000e+00  2.000000e+00  3.500000e+00  
waterfront     0.000000e+00  0.000000e+00  0.000000e+00  1.000000e+00  
view           0.000000e+00  0.000000e+00  0.000000e+00  4.000000e+00  
condition      3.000000e+00  3.000000e+00  4.000000e+00  5.000000e+00  
grade          7.000000e+00  7.000000e+00  8.000000e+00  1.300000e+01  
sqft_above     1.190000e+03  1.560000e+03  2.210000e+03  9.410000e+03  
yr_built       1.951000e+03  1.975000e+03  1.997000e+03  2.015000e+03  
yr_renovated   0.000000e+00  0.000000e+00  0.000000e+00  2.015000e+03  
zipcode        9.803300e+04  9.806500e+04  9.811800e+04  9.819900e+04  
lat            4.747110e+01  4.757180e+01  4.767800e+01  4.777760e+01  
long          -1.223280e+02 -1.222310e+02 -1.221250e+02 -1.213150e+02  
sqft_living15  1.490000e+03  1.840000e+03  2.360000e+03  6.210000e+03  
sqft_lot15     5.100000e+03  7.620000e+03  1.008300e+04  8.712000e+05

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
id               21597 non-null int64
date             21597 non-null object
price            21597 non-null float64
bedrooms         21597 non-null int64
bathrooms        21597 non-null float64
sqft_living      21597 non-null int64
sqft_lot         21597 non-null int64
floors           21597 non-null float64
waterfront       19221 non-null float64
view             21534 non-null float64
condition        21597 non-null int64
grade            21597 non-null int64
sqft_above       21597 non-null int64
sqft_basement    21597 non-null object
yr_built         21597 non-null int64
yr_renovated     17755 non-null float64
zipcode          21597 non-null int64
lat              21597 non-null float64
long             21597 non-null float64
sqft_living15    21597 non-null int64
sqft_lot15       21597 non-null int64
dtypes: float64(8), int64(11), object(2)
memory usage: 3.5+ MB


In [13]:
df.isna().any()

id               False
date             False
price            False
bedrooms         False
bathrooms        False
sqft_living      False
sqft_lot         False
floors           False
waterfront        True
view              True
condition        False
grade            False
sqft_above       False
sqft_basement    False
yr_built         False
yr_renovated      True
zipcode          False
lat              False
long             False
sqft_living15    False
sqft_lot15       False
dtype: bool

In [14]:
df.isna().sum()
#Lot of 

id                  0
date                0
price               0
bedrooms            0
bathrooms           0
sqft_living         0
sqft_lot            0
floors              0
waterfront       2376
view               63
condition           0
grade               0
sqft_above          0
sqft_basement       0
yr_built            0
yr_renovated     3842
zipcode             0
lat                 0
long                0
sqft_living15       0
sqft_lot15          0
dtype: int64

In [15]:
df.isna().sum().sum()

6281

We have 6,281 missing values broken down as follows: yr_renovates 3842, waterfront 2376, and view 63. We have to come up with a strategy to deal with them.

In [16]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

In [17]:
df.dtypes[df.dtypes == 'object']

date             object
sqft_basement    object
dtype: object

In [18]:
for feature in df.dtypes[df.dtypes == 'object'].index:
    print(feature)

date
sqft_basement


In [19]:
df.head(10)

id        date      price  bedrooms  bathrooms  sqft_living  \
0  7129300520  10/13/2014   221900.0         3       1.00         1180   
1  6414100192   12/9/2014   538000.0         3       2.25         2570   
2  5631500400   2/25/2015   180000.0         2       1.00          770   
3  2487200875   12/9/2014   604000.0         4       3.00         1960   
4  1954400510   2/18/2015   510000.0         3       2.00         1680   
5  7237550310   5/12/2014  1230000.0         4       4.50         5420   
6  1321400060   6/27/2014   257500.0         3       2.25         1715   
7  2008000270   1/15/2015   291850.0         3       1.50         1060   
8  2414600126   4/15/2015   229500.0         3       1.00         1780   
9  3793500160   3/12/2015   323000.0         3       2.50         1890   

   sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
0      5650     1.0         NaN   0.0     ...          7        1180   
1      7242     2.0         0.0   0.0     ...          7        2170   
2     10000     1.0         0.0   0.0     ...          6         770   
3      5000     1.0         0.0   0.0     ...          7        1050   
4      8080     1.0         0.0   0.0     ...          8        1680   
5    101930     1.0         0.0   0.0     ...         11        3890   
6      6819     2.0         0.0   0.0     ...          7        1715   
7      9711     1.0         0.0   NaN     ...          7        1060   
8      7470     1.0         0.0   0.0     ...          7        1050   
9      6560     2.0         0.0   0.0     ...          7        1890   

   sqft_basement yr_built  yr_renovated  zipcode      lat     long  \
0            0.0     1955           0.0    98178  47.5112 -122.257   
1          400.0     1951        1991.0    98125  47.7210 -122.319   
2            0.0     1933           NaN    98028  47.7379 -122.233   
3          910.0     1965           0.0    98136  47.5208 -122.393   
4            0.0     1987           0.0    98074  47.6168 -122.045   
5         1530.0     2001           0.0    98053  47.6561 -122.005   
6              ?     1995           0.0    98003  47.3097 -122.327   
7            0.0     1963           0.0    98198  47.4095 -122.315   
8          730.0     1960           0.0    98146  47.5123 -122.337   
9            0.0     2003           0.0    98038  47.3684 -122.031   

   sqft_living15  sqft_lot15  
0           1340        5650  
1           1690        7639  
2           2720        8062  
3           1360        5000  
4           1800        7503  
5           4760      101930  
6           2238        6819  
7           1650        9711  
8           1780        8113  
9           2390        7570  

[10 rows x 21 columns]

In [21]:
df[df['sqft_basement'] == '?']
# We 454 rows where df['sqft_basement'] == '?', we have to deal with those

id        date      price  bedrooms  bathrooms  sqft_living  \
6      1321400060   6/27/2014   257500.0         3       2.25         1715   
18       16000397   12/5/2014   189000.0         2       1.00         1200   
42     7203220400    7/7/2014   861990.0         5       2.75         3595   
79     1531000030   3/23/2015   720000.0         4       2.50         3450   
112    2525310310   9/16/2014   272500.0         3       1.75         1540   
115    3626039325  11/21/2014   740500.0         3       3.50         4380   
217    1909600046    7/3/2014   445838.0         3       2.50         2250   
309    3204800200    1/8/2015   665000.0         4       2.75         3320   
384     713500030   7/28/2014  1350000.0         5       3.50         4800   
508    5113400431    5/8/2014   615000.0         2       1.00         1540   
633    8832900550   9/12/2014   650000.0         3       2.50         2690   
645    1545804340    4/9/2015   240000.0         3       1.75         1760   
664    1775920350  11/24/2014   323000.0         3       1.00         1290   
671    5101405124   9/12/2014   435000.0         4       2.50         1700   
685    2346200030    1/5/2015   802541.0         5       2.75         2990   
755    5149300200    9/2/2014   316500.0         3       1.75         1600   
881    7749500370  10/21/2014   225000.0         4       2.25         1800   
890    1962200435  11/10/2014  1010000.0         4       1.00         1820   
995    8856004730   9/17/2014   199950.0         2       2.75         1590   
1073   9209900270    2/5/2015   515000.0         2       1.00         1060   
1105   2505500030    7/3/2014  1130000.0         4       2.50         3160   
1205    194000505    9/4/2014   651000.0         3       2.00         1940   
1354   3904940070   2/27/2015   643000.0         4       2.50         2270   
1356   5589300370    4/1/2015   282000.0         4       1.00         1200   
1366   3905000340   7/24/2014   672000.0         4       2.50         2440   
1476   2412600030   6/23/2014   235000.0         6       3.00         2180   
1558   1972201820  10/16/2014   610000.0         4       2.00         2130   
1600   3623500135   3/26/2015   800000.0         4       2.25         2350   
1662     65000260   8/20/2014   830000.0         3       2.50         3370   
1669   1785500132  11/18/2014   293000.0         3       1.00         1020   
...           ...         ...        ...       ...        ...          ...   
20275  9478550430   4/29/2015   316475.0         4       2.50         1740   
20333  7502800050    7/9/2014   659950.0         4       2.75         3510   
20510  2902200142    6/5/2014   584000.0         3       2.50         1480   
20553  8895800110    8/5/2014  1300000.0         5       4.00         3870   
20554  3879900750   9/10/2014   579000.0         2       2.50         1280   
20676  3274800505   4/24/2015   502000.0         3       2.50         1600   
20702  8850000517   7/31/2014   480000.0         3       2.50         1590   
20741  3022900070   9/29/2014   348000.0         3       2.00         2360   
20766  3013300685   3/18/2015   760000.0         4       3.25         2690   
20853  6130500120   4/17/2015   428000.0         3       2.50         1650   
20903  9578140180   6/11/2014   329950.0         3       2.50         2456   
20929  2568200120  12/15/2014   730000.0         5       2.75         2870   
20931  2767601750   8/15/2014   500000.0         3       1.50         1220   
20934  1689401526   3/23/2015   605000.0         3       2.50         1500   
20992  9274200322   8/20/2014   580000.0         3       2.50         1740   
21000   291310180   6/13/2014   379500.0         3       2.25         1410   
21017  7203230040  10/27/2014  1050000.0         5       3.25         4240   
21087  8564860130    2/2/2015   598992.0         5       3.50         3440   
21109  3438500250   6/23/2014   515000.0         5       3.25         2910   
21210  3278600680   6/27/2014   23500

In [22]:
df.tail(15)

id        date      price  bedrooms  bathrooms  sqft_living  \
21582  8956200760  10/13/2014   541800.0         4       2.50         3118   
21583  7202300110   9/15/2014   810000.0         4       3.00         3990   
21584   249000205  10/15/2014  1540000.0         5       3.75         4470   
21585  5100403806    4/7/2015   467000.0         3       2.50         1425   
21586   844000965   6/26/2014   224000.0         3       1.75         1500   
21587  7852140040   8/25/2014   507250.0         3       2.50         2270   
21588  9834201367   1/26/2015   429000.0         3       2.00         1490   
21589  3448900210  10/14/2014   610685.0         4       2.50         2520   
21590  7936000429   3/26/2015  1010000.0         4       3.50         3510   
21591  2997800021   2/19/2015   475000.0         3       2.50         1310   
21592   263000018   5/21/2014   360000.0         3       2.50         1530   
21593  6600060120   2/23/2015   400000.0         4       2.50         2310   
21594  1523300141   6/23/2014   402101.0         2       0.75         1020   
21595   291310100   1/16/2015   400000.0         3       2.50         1600   
21596  1523300157  10/15/2014   325000.0         2       0.75         1020   

       sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
21582      7866     2.0         NaN   2.0     ...          9        3118   
21583      7838     2.0         0.0   0.0     ...          9        3990   
21584      8088     2.0         0.0   0.0     ...         11        4470   
21585      1179     3.0         0.0   0.0     ...          8        1425   
21586     11968     1.0         NaN   0.0     ...          6        1500   
21587      5536     2.0         NaN   0.0     ...          8        2270   
21588      1126     3.0         0.0   0.0     ...          8        1490   
21589      6023     2.0         0.0   NaN     ...          9        2520   
21590      7200     2.0         0.0   0.0     ...          9        2600   
21591      1294     2.0         0.0   0.0     ...          8        1180   
21592      1131     3.0         0.0   0.0     ...          8        1530   
21593      5813     2.0         0.0   0.0     ...          8        2310   
21594      1350     2.0         0.0   0.0     ...          7        1020   
21595      2388     2.0         NaN   0.0     ...          8        1600   
21596      1076     2.0         0.0   0.0     ...          7        1020   

       sqft_basement yr_built  yr_renovated  zipcode      lat     long  \
21582            0.0     2014           0.0    98001  47.2931 -122.264   
21583            0.0     2003           NaN    98053  47.6857 -122.046   
21584            0.0     2008           0.0    98004  47.6321 -122.200   
21585            0.0     2008           0.0    98125  47.6963 -122.318   
21586            0.0     2014           0.0    98010  47.3095 -122.002   
21587            0.0     2003           0.0    98065  47.5389 -121.881   
21588            0.0     2014           0.0    98144  47.5699 -122.288   
21589            0.0     2014           0.0    98056  47.5137 -122.167   
21590          910.0     2009           0.0    98136  47.5537 -122.398   
21591          130.0     2008           0.0    98116  47.5773 -122.409   
21592            0.0     2009           0.0    98103  47.6993 -122.346   
21593            0.0     2014           0.0    98146  47.5107 -122.362   
21594            0.0     2009           0.0    98144  47.5944 -122.299   
21595            0.0     2004           0.0    98027  47.5345 -122.069   
21596            0.0     2008           0.0    98144  47.5941 -122.299   

       sqft_living15  sqft_lot15  
21582           2673        6500  
21583           3370        6814  
21584           2780        8964  
21585           1285        1253  
21586           1320       11303  
21587           2270        5731  
21588           1400        1230  
21589           2520        6023  
21590           2050        6200  
21591         

In [37]:
df[df['sqft_basement'] == '0.0']
# 12826 rows 

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7129300520  10/13/2014   221900.0         3       1.00         1180   
2      5631500400   2/25/2015   180000.0         2       1.00          770   
4      1954400510   2/18/2015   510000.0         3       2.00         1680   
7      2008000270   1/15/2015   291850.0         3       1.50         1060   
9      3793500160   3/12/2015   323000.0         3       2.50         1890   
12      114101516   5/28/2014   310000.0         3       1.00         1430   
13     6054650070   10/7/2014   400000.0         3       1.75         1370   
14     1175000570   3/12/2015   530000.0         5       2.00         1810   
16     1875500060   7/31/2014   395000.0         3       2.00         1890   
17     6865200140   5/29/2014   485000.0         4       1.00         1600   
19     7983200060   4/24/2015   230000.0         3       1.00         1250   
22     7137970340    7/3/2014   285000.0         5       2.50         2270   
23     8091400200   5/16/2014   252700.0         2       1.50         1070   
24     3814700200  11/20/2014   329000.0         3       2.25         2450   
25     1202000200   11/3/2014   233000.0         3       2.00         1710   
27     3303700376   12/1/2014   667000.0         3       1.00         1400   
29     1873100390    3/2/2015   719000.0         4       2.50         2570   
30     8562750320  11/10/2014   580500.0         3       2.50         2320   
31     2426039314   12/1/2014   280000.0         2       1.50         1190   
33     7589200193  11/10/2014   535000.0         3       1.00         1090   
37     2768000400  12/30/2014   640000.0         4       2.00         2360   
39     2078500320   6/20/2014   605000.0         4       2.50         2620   
40     5547700270   7/15/2014   625000.0         4       2.50         2570   
43     9270200160  10/28/2014   685000.0         3       1.00         1570   
45     8035350320   7/18/2014   488000.0         3       2.50         3160   
46     8945200830   3/25/2015   210490.0         3       1.00          990   
47     4178300310   7/16/2014   785000.0         4       2.50         2290   
48     9215400105   4/28/2015   450000.0         3       1.75         1250   
50     5245600105   9/16/2014   228000.0         3       1.00         1190   
55     9822700295   5/12/2014   885000.0         4       2.50         2830   
...           ...         ...        ...       ...        ...          ...   
21553  6021503705  10/15/2014   329000.0         2       2.50          980   
21556  6056111067    7/7/2014   230000.0         3       1.75         1140   
21558  1931300090    5/7/2014   610950.0         3       3.00         1680   
21559  9578500790  11/11/2014   399950.0         3       2.50         3087   
21562   567000385   6/23/2014   362500.0         2       1.50          940   
21564  7853420110   10/3/2014   594866.0         3       3.00         2780   
21565  7853420110    5/4/2015   625000.0         3       3.00         2780   
21567  2025049203   6/10/2014   399950.0         2       1.00          710   
21569  3832050760   8/28/2014   270000.0         3       2.50         1870   
21570  2767604724  10/15/2014   505000.0         2       2.50         1430   
21571  6632300207    3/5/2015   385000.0         3       2.50         1520   
21573  7570050450   9/10/2014   347500.0         3       2.50         2540   
21575  4140940150   10/2/2014   572000.0         4       2.75         2770   
21576  1931300412   4/16/2015   475000.0         3       2.25         1190   
21577  8672200110   3/17/2015  1090000.0         5       3.75         4170   
21578  5087900040  10/17/2014   350000.0         4       2.75         2500   
21580  7502800100   8/13/2014   679950.0         5       2.75         3600   
21582  8956200760  10/13/2014   541800.0         4       2.50         3118   
21583  7202300110   9/15/2014   810000.0         4       3.00         3990   
21584   249000205  10/15/2014  154000

In [31]:
df[df['waterfront'] == 0.0]
# 19075 rows

id        date      price  bedrooms  bathrooms  sqft_living  \
1      6414100192   12/9/2014   538000.0         3       2.25         2570   
2      5631500400   2/25/2015   180000.0         2       1.00          770   
3      2487200875   12/9/2014   604000.0         4       3.00         1960   
4      1954400510   2/18/2015   510000.0         3       2.00         1680   
5      7237550310   5/12/2014  1230000.0         4       4.50         5420   
6      1321400060   6/27/2014   257500.0         3       2.25         1715   
7      2008000270   1/15/2015   291850.0         3       1.50         1060   
8      2414600126   4/15/2015   229500.0         3       1.00         1780   
9      3793500160   3/12/2015   323000.0         3       2.50         1890   
11     9212900260   5/27/2014   468000.0         2       1.00         1160   
12      114101516   5/28/2014   310000.0         3       1.00         1430   
13     6054650070   10/7/2014   400000.0         3       1.75         1370   
14     1175000570   3/12/2015   530000.0         5       2.00         1810   
15     9297300055   1/24/2015   650000.0         4       3.00         2950   
16     1875500060   7/31/2014   395000.0         3       2.00         1890   
17     6865200140   5/29/2014   485000.0         4       1.00         1600   
18       16000397   12/5/2014   189000.0         2       1.00         1200   
19     7983200060   4/24/2015   230000.0         3       1.00         1250   
20     6300500875   5/14/2014   385000.0         4       1.75         1620   
21     2524049179   8/26/2014  2000000.0         3       2.75         3050   
22     7137970340    7/3/2014   285000.0         5       2.50         2270   
24     3814700200  11/20/2014   329000.0         3       2.25         2450   
25     1202000200   11/3/2014   233000.0         3       2.00         1710   
26     1794500383   6/26/2014   937000.0         3       1.75         2450   
27     3303700376   12/1/2014   667000.0         3       1.00         1400   
28     5101402488   6/24/2014   438000.0         3       1.75         1520   
29     1873100390    3/2/2015   719000.0         4       2.50         2570   
30     8562750320  11/10/2014   580500.0         3       2.50         2320   
31     2426039314   12/1/2014   280000.0         2       1.50         1190   
32      461000390   6/24/2014   687500.0         4       1.75         2330   
...           ...         ...        ...       ...        ...          ...   
21559  9578500790  11/11/2014   399950.0         3       2.50         3087   
21561  3881900317   1/23/2015   579000.0         4       3.25         1900   
21563  7011201004   5/29/2014   645000.0         3       3.25         1730   
21564  7853420110   10/3/2014   594866.0         3       3.00         2780   
21565  7853420110    5/4/2015   625000.0         3       3.00         2780   
21566  3052700432  11/12/2014   490000.0         3       2.25         1500   
21568   952006823   12/2/2014   380000.0         3       2.50         1260   
21569  3832050760   8/28/2014   270000.0         3       2.50         1870   
21570  2767604724  10/15/2014   505000.0         2       2.50         1430   
21571  6632300207    3/5/2015   385000.0         3       2.50         1520   
21572  2767600688  11/13/2014   414500.0         2       1.50         1210   
21573  7570050450   9/10/2014   347500.0         3       2.50         2540   
21574  7430200100   5/14/2014  1220000.0         4       3.50         4910   
21575  4140940150   10/2/2014   572000.0         4       2.75         2770   
21576  1931300412   4/16/2015   475000.0         3       2.25         1190   
21577  8672200110   3/17/2015  1090000.0         5       3.75         4170   
21579  1972201967  10/31/2014   520000.0         2       2.25         1530   
21580  7502800100   8/13/2014   679950.0         5       2.75         3600   
21581   191100405   4/21/2015  1580000.0         4       3.25         3410   
21583  7202300110   9/15/2014   81000

In [32]:
df[df['view'] == 0.0]
#  19422 rows

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7129300520  10/13/2014   221900.0         3       1.00         1180   
1      6414100192   12/9/2014   538000.0         3       2.25         2570   
2      5631500400   2/25/2015   180000.0         2       1.00          770   
3      2487200875   12/9/2014   604000.0         4       3.00         1960   
4      1954400510   2/18/2015   510000.0         3       2.00         1680   
5      7237550310   5/12/2014  1230000.0         4       4.50         5420   
6      1321400060   6/27/2014   257500.0         3       2.25         1715   
8      2414600126   4/15/2015   229500.0         3       1.00         1780   
9      3793500160   3/12/2015   323000.0         3       2.50         1890   
10     1736800520    4/3/2015   662500.0         3       2.50         3560   
11     9212900260   5/27/2014   468000.0         2       1.00         1160   
12      114101516   5/28/2014   310000.0         3       1.00         1430   
13     6054650070   10/7/2014   400000.0         3       1.75         1370   
14     1175000570   3/12/2015   530000.0         5       2.00         1810   
16     1875500060   7/31/2014   395000.0         3       2.00         1890   
17     6865200140   5/29/2014   485000.0         4       1.00         1600   
18       16000397   12/5/2014   189000.0         2       1.00         1200   
19     7983200060   4/24/2015   230000.0         3       1.00         1250   
20     6300500875   5/14/2014   385000.0         4       1.75         1620   
22     7137970340    7/3/2014   285000.0         5       2.50         2270   
23     8091400200   5/16/2014   252700.0         2       1.50         1070   
24     3814700200  11/20/2014   329000.0         3       2.25         2450   
25     1202000200   11/3/2014   233000.0         3       2.00         1710   
26     1794500383   6/26/2014   937000.0         3       1.75         2450   
27     3303700376   12/1/2014   667000.0         3       1.00         1400   
28     5101402488   6/24/2014   438000.0         3       1.75         1520   
29     1873100390    3/2/2015   719000.0         4       2.50         2570   
30     8562750320  11/10/2014   580500.0         3       2.50         2320   
31     2426039314   12/1/2014   280000.0         2       1.50         1190   
32      461000390   6/24/2014   687500.0         4       1.75         2330   
...           ...         ...        ...       ...        ...          ...   
21564  7853420110   10/3/2014   594866.0         3       3.00         2780   
21565  7853420110    5/4/2015   625000.0         3       3.00         2780   
21566  3052700432  11/12/2014   490000.0         3       2.25         1500   
21567  2025049203   6/10/2014   399950.0         2       1.00          710   
21568   952006823   12/2/2014   380000.0         3       2.50         1260   
21569  3832050760   8/28/2014   270000.0         3       2.50         1870   
21570  2767604724  10/15/2014   505000.0         2       2.50         1430   
21571  6632300207    3/5/2015   385000.0         3       2.50         1520   
21572  2767600688  11/13/2014   414500.0         2       1.50         1210   
21573  7570050450   9/10/2014   347500.0         3       2.50         2540   
21574  7430200100   5/14/2014  1220000.0         4       3.50         4910   
21575  4140940150   10/2/2014   572000.0         4       2.75         2770   
21576  1931300412   4/16/2015   475000.0         3       2.25         1190   
21578  5087900040  10/17/2014   350000.0         4       2.75         2500   
21579  1972201967  10/31/2014   520000.0         2       2.25         1530   
21580  7502800100   8/13/2014   679950.0         5       2.75         3600   
21581   191100405   4/21/2015  1580000.0         4       3.25         3410   
21583  7202300110   9/15/2014   810000.0         4       3.00         3990   
21584   249000205  10/15/2014  1540000.0         5       3.75         4470   
21585  5100403806    4/7/2015   46700

In [35]:
df[df['yr_renovated'] == 0.0]


id        date      price  bedrooms  bathrooms  sqft_living  \
0      7129300520  10/13/2014   221900.0         3       1.00         1180   
3      2487200875   12/9/2014   604000.0         4       3.00         1960   
4      1954400510   2/18/2015   510000.0         3       2.00         1680   
5      7237550310   5/12/2014  1230000.0         4       4.50         5420   
6      1321400060   6/27/2014   257500.0         3       2.25         1715   
7      2008000270   1/15/2015   291850.0         3       1.50         1060   
8      2414600126   4/15/2015   229500.0         3       1.00         1780   
9      3793500160   3/12/2015   323000.0         3       2.50         1890   
10     1736800520    4/3/2015   662500.0         3       2.50         3560   
11     9212900260   5/27/2014   468000.0         2       1.00         1160   
13     6054650070   10/7/2014   400000.0         3       1.75         1370   
14     1175000570   3/12/2015   530000.0         5       2.00         1810   
15     9297300055   1/24/2015   650000.0         4       3.00         2950   
16     1875500060   7/31/2014   395000.0         3       2.00         1890   
17     6865200140   5/29/2014   485000.0         4       1.00         1600   
18       16000397   12/5/2014   189000.0         2       1.00         1200   
19     7983200060   4/24/2015   230000.0         3       1.00         1250   
20     6300500875   5/14/2014   385000.0         4       1.75         1620   
21     2524049179   8/26/2014  2000000.0         3       2.75         3050   
22     7137970340    7/3/2014   285000.0         5       2.50         2270   
24     3814700200  11/20/2014   329000.0         3       2.25         2450   
25     1202000200   11/3/2014   233000.0         3       2.00         1710   
27     3303700376   12/1/2014   667000.0         3       1.00         1400   
29     1873100390    3/2/2015   719000.0         4       2.50         2570   
30     8562750320  11/10/2014   580500.0         3       2.50         2320   
31     2426039314   12/1/2014   280000.0         2       1.50         1190   
32      461000390   6/24/2014   687500.0         4       1.75         2330   
33     7589200193  11/10/2014   535000.0         3       1.00         1090   
34     7955080270   12/3/2014   322500.0         4       2.75         2060   
36     9435300030   5/28/2014   550000.0         4       1.00         1660   
...           ...         ...        ...       ...        ...          ...   
21560  9253900271    1/7/2015  3570000.0         5       4.50         4850   
21561  3881900317   1/23/2015   579000.0         4       3.25         1900   
21562   567000385   6/23/2014   362500.0         2       1.50          940   
21563  7011201004   5/29/2014   645000.0         3       3.25         1730   
21564  7853420110   10/3/2014   594866.0         3       3.00         2780   
21566  3052700432  11/12/2014   490000.0         3       2.25         1500   
21567  2025049203   6/10/2014   399950.0         2       1.00          710   
21568   952006823   12/2/2014   380000.0         3       2.50         1260   
21569  3832050760   8/28/2014   270000.0         3       2.50         1870   
21570  2767604724  10/15/2014   505000.0         2       2.50         1430   
21571  6632300207    3/5/2015   385000.0         3       2.50         1520   
21572  2767600688  11/13/2014   414500.0         2       1.50         1210   
21573  7570050450   9/10/2014   347500.0         3       2.50         2540   
21574  7430200100   5/14/2014  1220000.0         4       3.50         4910   
21578  5087900040  10/17/2014   350000.0         4       2.75         2500   
21580  7502800100   8/13/2014   679950.0         5       2.75         3600   
21582  8956200760  10/13/2014   541800.0         4       2.50         3118   
21584   249000205  10/15/2014  1540000.0         5       3.75         4470   
21585  5100403806    4/7/2015   467000.0         3       2.50         1425   
21586   844000965   6/26/2014   22400

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')